hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieMetadataSync.java

```diff
@@ -245,90 +248,166 @@ public void run() throws Exception {
+      try {
+        HoodieReplaceCommitMetadata commitMetadata = HoodieReplaceCommitMetadata.fromBytes(
+            sourceTableMetaClient.getCommitsTimeline().getInstantDetails(replaceCommit).get(), HoodieReplaceCommitMetadata.class);
+        Map<String, List<String>> partitionsToReplacedFileGroups = commitMetadata.getPartitionToReplaceFileIds();
+        for (Map.Entry<String, List<String>> entry : partitionsToReplacedFileGroups.entrySet()) {
+          String partition = entry.getKey();
+          List<String> replacedFiles = entry.getValue();
+          totalPartitionToReplacedFiles.computeIfAbsent(partition, k -> new ArrayList<>());
+          totalPartitionToReplacedFiles.get(partition).addAll(replacedFiles);
+        }
+      } catch (IOException e) {
+        throw new HoodieException("Failed to deserialize instant " + replaceCommit.getTimestamp(), e);
+      }
+    });
 
-          try (HoodieTableMetadataWriter hoodieTableMetadataWriter =
-                   (HoodieTableMetadataWriter) sparkTable.getMetadataWriter(commitTime).get()) {
-            hoodieTableMetadataWriter.update(tgtReplaceCommitMetadata, hoodieSparkEngineContext.emptyHoodieData(), commitTime);
-          }
-          // add metadata sync checkpoint info
-          tgtReplaceCommitMetadata.addMetadata(SyncMetadata.TABLE_SYNC_METADATA, syncMetadata.toJson());
-          commitMetadataInBytes = Option.of(tgtReplaceCommitMetadata.toJsonString().getBytes(StandardCharsets.UTF_8));
-        } else if (instant.getAction().equals(HoodieTimeline.CLEAN_ACTION)) {
-          HoodieCleanMetadata srcCleanMetadata = TimelineMetadataUtils.deserializeHoodieCleanMetadata(
-              sourceTableMetaClient.getCommitsTimeline().getInstantDetails(instant).get());
-          HoodieCleanMetadata tgtCleanMetadata = reconstructHoodieCleanCommitMetadata(srcCleanMetadata,
-              writeConfig, hoodieSparkEngineContext, targetTableMetaClient);
-          //HoodieCleanMetadata tgtCleanMetadata = buildHoodieCleanMetadata(srcCleanMetadata, commitTime);
-          try (HoodieTableMetadataWriter hoodieTableMetadataWriter =
-                   (HoodieTableMetadataWriter) sparkTable.getMetadataWriter(commitTime).get()) {
-            hoodieTableMetadataWriter.update(tgtCleanMetadata, commitTime);
-          }
+    replaceCommitMetadata.setPartitionToReplaceFileIds(totalPartitionToReplacedFiles);
 
-          commitMetadataInBytes = TimelineMetadataUtils.serializeCleanMetadata(tgtCleanMetadata);
-        }
+    replaceCommitMetadata.addMetadata("schema", schema.toString());
+    replaceCommitMetadata.setOperationType(WriteOperationType.BOOTSTRAP);
+    replaceCommitMetadata.setCompacted(false);
+    return replaceCommitMetadata;
+  }
 
-        targetTableMetaClient
-            .reloadActiveTimeline()
-            .saveAsComplete(new HoodieInstant(true, instant.getAction(), commitTime), commitMetadataInBytes);
-      }
+  private void runArchiver(
```

Comment:
java docs

Reply:
don't we need a lock here? 
what if two concurrent writer tries to archive the target table concurrently